### PR TITLE
Create update landing page authors

### PR DESCRIPTION
**Pull request by bpowers24**
- The goal was to create a new function that allows us to update authors on ESS-DIVE's landing pages. 
- Potential use cases include adding authors to manuscript data package landing pages when a pre-publication DOI is needed before the data are finalized and updating study data package landing pages to reflect new authors added due to newly available data types. 

| Objectives | Results |
|--------|--------|
| Create a function that uses the API to update authors on an already existing ESS-DIVE landing page | See `update_landing_page_authors()` function |
| Be able to parse the ESS-DIVE metadata file to get author names for a given data package | See `get_authors_from_essdive_metadata()` function |
| Based on an author list, pull info from the internal author spreadsheet to get ORCIDs, emails, and institution names | See `get_author_spreadsheet_info()` function | 
Create a "buffet" style script that combines these new author functions together for easy reuse | See `update_landing_page.R` script |
Combine the author and coordinate update functions together |  See `update_landing_page.R` script |
Write documentation for the author and coordinate scripts that anyone (even those outside our project) can refer to and use | See the readmes in the `update_ESS-DIVE_landing_page` directory | 
Stress test the author functions to make sure they pass standard and edge cases and throw errors for common mistakes | See `test-update_landing_page_authors.R` script for all the tests that currently pass. Note: I only wrote tests for the two internal functions.

### CODE REVIEW

**Comment by bpowers24** [comment]
The test file is kinda unruly, so here's a summary of the things I was checking for: 
image

The numbers in the brackets are all the different combinations for these 3 types of name formats: 
1. first middle last; 
3. first last; 
4. 3. last

File: Data_Package_ESS-DIVE/update_ESS-DIVE_landing_page/README_update_ESS-DIVE_landing_page_authors.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+## Prepare authors
+Create a data frame with 5 (required) columns. 
+
+| first_name                                                                          | last_name                                               | orcid                                                                   | affiliation                                                                   | email                                                    |
+| ----------------------------------------------------------------------------------- | ------------------------------------------------------- | ----------------------------------------------------------------------- | ----------------------------------------------------------------------------- | -------------------------------------------------------- |
+| `<chr>` The author's first name. Add middle names here too. This field is required. | `<chr>` The author's last name. This field is required. | `<num>` The author's ORICD number. If not applicable, leave empty (NA). | The author's institution or affiliation. If not applicable, leave empty (NA). | The author's email. If not applicable, leave empty (NA). |
```
Comment (brieanne-forbes) [inline Data_Package_ESS-DIVE/update_ESS-DIVE_landing_page/README_update_ESS-DIVE_landing_page_authors.md:9]:
```suggestion
| `<chr>` The author's first name and (if applicable) middle names/initial. This field is required. | `<chr>` The author's last name. This field is required. | `<num>` The author's ORICD number. If not applicable, leave empty (NA). | The author's institution or affiliation. If not applicable, leave empty (NA). | The author's email. If not applicable, leave empty (NA). |
```

File: Data_Package_ESS-DIVE/update_ESS-DIVE_landing_page/README_update_ESS-DIVE_landing_page_authors.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  last_name = c("Johnson", "Lee", "Smith"),
+  orcid = c("0000-0000-0000-1111", NA, "0000-0000-0000-2222"),
+  affiliation = c("University of Washington", "Pacific Northwest National Laboratory", NA),
+  email = c("alice.johnson@uw.edu", "bob.lee@pnnl.gov", NA)
+)
+```
+
+## About the function
+The `update_landing_page_authors()` function has 4 (required) arguments: 
+1. `api_token`: This is your personal token. Sign into to ESS-DIVE > My Settings > Authentication Token > Copy Token
```
Comment (brieanne-forbes) [inline Data_Package_ESS-DIVE/update_ESS-DIVE_landing_page/README_update_ESS-DIVE_landing_page_authors.md:25]:
```suggestion
1. `api_token`: This is your personal token. Sign into to ESS-DIVE > My Settings > Authentication Token > Copy Token. It is recommended to add this into the console (rather than the script) so that your personal API token does not get saved within the script. 
```

**Review by brieanne-forbes — APPROVED** [review]
It would be nice if you distinguished the functions from the script. For dd/flmd for example, you have the functions in a subfolder. Or you could just add "functions" to the file name.
